### DISCUSSION TITLE
Why there is no documentation such as the old project?

### DISCUSSION THREAD

**Discussion by kikutano** [dotnet/eShop#262] [Q&A] [answered]
Hello to everyone,
First of all, thanks for this project. The only think that I don't understand is why there is no documentation such as the old source code https://github.com/dotnet-architecture/eShopOnContainers. I cannot find any reference about design choice, code and other things. I'm missing something?

Thanks!

**Comment by ASADeveloper** [comment]
It also seems strange to me that the admins refer to another project eShopOnAzure for the documentation, while it also has no documentation!
Personally, I think this example and the code is quite simple. What I find interesting is the choices and the reasoning behind them. Without explanation of the technical choices, this is a plain code repository that has little educational value.

**Comment by imregyCG** [comment]
Have you looked at this?
https://dotnet.microsoft.com/en-us/download/e-book/microservices-architecture/pdf

**Reply by ASADeveloper** [reply]
Did you? ;-)

**Comment by kikutano** [comment]
> Have you looked at this? https://dotnet.microsoft.com/en-us/download/e-book/microservices-architecture/pdf

This is the documentation for the old repository.

**Comment by kuklei** [comment]
https://github.com/dotnet-architecture/eShopOnContainers/wiki this should be ported somehow. I'm not sure on the differences between the projects as I just discovered it, but it's a valid point.

**Comment by davidfowl** [comment, marked as answer]
We're working on a new book and new docs. We still want to make more code changes before we do that though. Hopefully you'll be able to see some progress soon.

**Reply by HelgiBergmann** [reply]
Can't wait, but until that happen, let's read the code :)

**Reply by thejazz1991** [reply]
Al menos si existiara la posibilida de una comunidad donde pudieramos recibir apoyo estaría bien en lo que se crea la documatación.

**Reply by numa-alset** [reply]
we appreciate your time and effort but if you could post whatever is done with the book so we could start learning or give us date of publish so we could arrange for that

**Reply by hesam-javadi** [reply]
A year has passed! Do you have any updates on this?

**Comment by HelgiBergmann** [comment]
Any updates?

**Comment by Jyothish-POC** [comment]
Any update on this?

**Comment by mike-geiger** [comment]
It has been nearly two years, and this has yet to materialize. Have these plans been abandoned? Will there be any guidance documentation going forward?

**Reply by OlehObodinWork** [reply]
To be honest it easy to get into project using Copilot

**Reply by mike-geiger** [reply]
That’s not really an answer is it. Can you elaborate how copilot is a replacement for the previous ebook?

**Reply by OlehObodinWork** [reply]
Copilot will structure and explain how parts are working. Also Aspire really helpful tool for help figure out what is going on more clearly. So don't be lazy, read the code.

**Reply by hollowdrutt** [reply]
The ".NET Microservices: Architecture for Containerized .NET Applications" which used "eShopOnContainers" as reference application is a 350 pages eBook, filled with explanations and reasonings behind choices made in the application, general design patterns, links to more resources on the subjects etc.
I don't think Copilots guesses about the source code will be as accurate and informative as such a book written and reviewed by subject matter experts. And I don't think reading the book is lazier than asking Copilot to explain code.

**Reply by rlarno** [reply]
For now, until @davidfowl unleashes his own AI skillzzz (maybe an AspireFridays topic, be sure to set up the bingo):
DeepWiki should get people going on understanding the basics of the architecture. At least the **HOW**
https://deepwiki.com/dotnet/eShop

I do agree the **WHY** is a lot more important and more interesting to know and learn.

Would be nice if the team would use ADR's in md format (reverse engineered from the e-book?), it could help future updates and better 'AI-Auto-Docs'.

**Reply by scottyinthematrix** [reply]
this seems helpful, thank you!
